source/extensions/listener_managers/listener_manager/connection_handler_impl.cc

```diff
@@ -290,8 +289,10 @@ void ConnectionHandlerImpl::setListenerRejectFraction(UnitFloat reject_fraction)
 Network::InternalListenerOptRef
 ConnectionHandlerImpl::findByAddress(const Network::Address::InstanceConstSharedPtr& address) {
   ASSERT(address->type() == Network::Address::Type::EnvoyInternal);
-  if (auto listener_it =
-          internal_listener_map_by_address_.find(address->envoyInternalAddress()->addressId());
+  Network::Address::InstanceConstSharedPtr internal_instance =
+      std::make_shared<Network::Address::EnvoyInternalInstance>(
+          address->envoyInternalAddress()->addressId());
```

Comment:
random thought, feel free to ignore: a heterogeneous lookup with value address instead of const-shared-ptr address would be slightly faster.

Reply:
OK, I don't remember off-hand the magic, but I recall that absl hash maps can accept string_view and string at the same time without a temporary.